dbr/test_app/tests/test_business_rules_plugin.py

```diff
@@ -0,0 +1,184 @@
+        self.assertDictEqual(actual, expected)
+
+    def test_should_execute_rules_1(self):
+        # GIVEN
+        product_for_sale = mommy.make(Products, current_inventory=21, price=200)
+        product_not_for_sale_1 = mommy.make(Products, current_inventory=20, price=200)
+        product_not_for_sale_2 = mommy.make(Products, current_inventory=21, price=200)
+
+        product_orders_for_sale = mommy.make(
+            ProductOrder,
+            product=product_for_sale,
+            expiration_date=timezone.now()+timezone.timedelta(days=4)
+        )
+        product_orders_not_for_sale_1 = mommy.make(
+            ProductOrder,
+            product=product_not_for_sale_1,
+            expiration_date=timezone.now() + timezone.timedelta(days=4)
+        )
+        product_orders_not_for_sale_2 = mommy.make(
+            ProductOrder,
+            product=product_not_for_sale_2,
+            expiration_date=timezone.now() + timezone.timedelta(days=5)
+        )
+
+        ProductBusinessRule.generate()
+        ProductBusinessRule.save_rules(self._get_rules())
+        # WHEN
+        for product in Products.objects.all():
+            ProductBusinessRule.run_all(product)
+        # THEN
+        product_for_sale.refresh_from_db()
+        product_not_for_sale_1.refresh_from_db()
+        product_not_for_sale_2.refresh_from_db()
+        self.assertEqual(product_for_sale.price, 150)
+        self.assertEqual(product_not_for_sale_1.price, 200)
+        self.assertEqual(product_not_for_sale_2.price, 200)
+
+    @patch.object(
+        timezone,
+        'now',
+        Mock(return_value=datetime.datetime(2015, 6, 1, tzinfo=timezone.utc))
+    )
+    def test_should_execute_rules_2(self):
+        # GIVEN
+        product_for_order = mommy.make(Products, current_inventory=4, price=200)
+        product_not_for_order = mommy.make(Products, current_inventory=5, price=200)
+
+        product_orders_for_order = mommy.make(
+            ProductOrder,
+            product=product_for_order,
+            expiration_date=datetime.datetime.now() + timezone.timedelta(days=4)
+        )
+        product_orders_not_for_order = mommy.make(
+            ProductOrder,
+            product=product_not_for_order,
+            expiration_date=datetime.datetime.now() + timezone.timedelta(days=4)
+        )
+
+        self.assertFalse(
+            ProductOrder.objects.filter(
+                product_id=product_orders_for_order.id,
+                quantity=40
+            ).exists()
+        )
+        self.assertFalse(
+            ProductOrder.objects.filter(
+                product_id=product_not_for_order.id,
+                quantity=40
+            ).exists()
+        )
+
+        ProductBusinessRule.generate()
+        ProductBusinessRule.save_rules(self._get_rules())
+        # WHEN
+        for product in Products.objects.all():
+            ProductBusinessRule.run_all(product)
+        # THEN
+        self.assertTrue(
```

Comment:
These 2 assertions were used at least 3 times in test cases. Maybe you can move them to some method or create custom assert?